src/components/SearchBar.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { view , store } from 'react-easy-state';
+import { BrowserRouter as Router, Redirect, Link } from "react-router-dom"
+import './Searchbar.css';
+
+interface SearchProps {
+    type:string,
+    placeholder?:string
+}
+
+interface SearchState {
+    searchValue?: string,
+    redirect?: boolean
+}
+
+const state = store<SearchState>({})
+
+class SearchBar extends React.Component<SearchProps>{
+    checkSubmit = (e:any) => {
+        if(e && e.keyCode == 13) {
+            state.redirect = true;
+        }
+    }
+
+    onChange = (e:any) => {
+        state.searchValue = e.target.value
+        console.log(state.searchValue)
+    }
+    render(){
+        return(
+            <>
+                {state.redirect === true &&
```

Comment:
Better to move big sections of logic like this in to `if` statements outside your return statement.